VmAgent.Core/Interfaces/IFileSystemOperations.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace VmAgent.Core.Interfaces
+{
+    public interface IFileSystemOperations
```

Comment:
Public documentation missing.  At least having a summary of why this classes is used.